ui/src/components/NodePartitionTable.test.js

```diff
@@ -0,0 +1,150 @@
+          status: {
+            inhibitedBy: [],
+            silencedBy: [],
+            state: 'active',
+          },
+          updatedAt: '2021-01-29T07:36:11.363Z',
+          generatorURL:
+            'http://prometheus-operator-prometheus.metalk8s-monitoring:9090/graph?g0.expr=%28node_filesystem_avail_bytes%7Bfstype%21%3D%22%22%2Cjob%3D%22node-exporter%22%7D+%2F+node_filesystem_size_bytes%7Bfstype%21%3D%22%22%2Cjob%3D%22node-exporter%22%7D+%2A+100+%3C+5+and+node_filesystem_readonly%7Bfstype%21%3D%22%22%2Cjob%3D%22node-exporter%22%7D+%3D%3D+0%29&g0.tab=1',
+          labels: {
+            alertname: 'NodeFilesystemAlmostOutOfSpace',
+            container: 'node-exporter',
+            device: '/dev/vdc',
+            endpoint: 'metrics',
+            fstype: 'xfs',
+            instance: '192.168.1.29:9100',
+            job: 'node-exporter',
+            mountpoint: '/mnt/testpart',
+            namespace: 'metalk8s-monitoring',
+            pod: 'prometheus-operator-prometheus-node-exporter-wk86s',
+            prometheus: 'metalk8s-monitoring/prometheus-operator-prometheus',
+            service: 'prometheus-operator-prometheus-node-exporter',
+            severity: 'warning',
+          },
+        },
+      ];
+      return res(ctx.json(alerts));
+    },
+  ),
+);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+test('the partition table', async () => {
+  // Setup
+
+  // use fake timers to let react query retry immediately after promise failure
+  jest.useFakeTimers();
+  initializeProm('http://192.168.1.18:8443/api/prometheus');
+  initializeAM('http://192.168.1.18:8443/api/alertmanager');
+
+  const { getByLabelText } = render(
+    <NodePartitionTable instanceIP={'192.168.1.29'} />,
+  );
+  expect(getByLabelText('loading')).toBeInTheDocument();
+
+  // Exercise
+  await waitForLoadingToFinish();
+
+  // Verify
+  expect(screen.getByLabelText('status warning')).toBeInTheDocument();
+  expect(screen.getByLabelText('percentage')).toBeInTheDocument();
+});
+
+test('handles server error', async () => {
+  // S
+  jest.useFakeTimers();
+  initializeProm('http://192.168.1.18:8443/api/prometheus');
+  initializeAM('http://192.168.1.18:8443/api/alertmanager');
+  // override the default route with error status
+  server.use(
+    rest.get(
+      'http://192.168.1.18:8443/api/prometheus/api/v1/query',
+      (req, res, ctx) => {
+        return res(ctx.status(500));
+      },
+    ),
+    rest.get(
+      'http://192.168.1.18:8443/api/alertmanager/api/v2/alerts',
+      (req, res, ctx) => {
+        return res(ctx.status(500));
+      },
+    ),
+  );
+  const { getByLabelText } = render(
+    <NodePartitionTable instanceIP={'192.168.1.29'} />,
+  );
+  expect(getByLabelText('loading')).toBeInTheDocument();
+
+  // E
+  await waitForLoadingToFinish();
+
+  // V
+  expect(
+    screen.getByText('System partitions request has failed.'),
+  ).toBeInTheDocument();
+});
```

Comment:
You may want to move this in a describe block to explicitly limit the scope of `beforeAll`, `afterEach` and `afterAll` functions to those tests